src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -517,19 +523,36 @@ private static T GetDeclarationSyntax<T>( ISymbol symbol )
 			return decl;
 		}
 
-		private MutabilityInspectionResult InspectMemberRecursive(
+		private MutabilityInspectionResult InspectMember(
 			ISymbol symbol,
 			HashSet<ITypeSymbol> typeStack
 		) {
-			// if the member is audited or unaudited, ignore it
-			if( Attributes.Mutability.Audited.IsDefined( symbol ) ) {
-				return MutabilityInspectionResult.NotMutable();
-			}
-			if( Attributes.Mutability.Unaudited.IsDefined( symbol ) ) {
-				string unauditedReason = BecauseHelpers.GetUnauditedReason( symbol );
-				return MutabilityInspectionResult.NotMutable( ImmutableHashSet.Create( unauditedReason ) );
+			// if the member is audited or unaudited, check if valid
+			if( Attributes.Mutability.IsDefined( symbol ) ) {
+
+				var annotedAttributes = Attributes.Mutability.GetAllAttributeSyntax( symbol );
+				var annotatedInspectionResult = InspectMemberRecursive( symbol, typeStack );
+
+				var annotatedResult = MutabilityInspectionResult.Annotated(
+					annotatedInspectionResult,
+					annotedAttributes
+				);
+
+				if( Attributes.Mutability.Unaudited.IsDefined( symbol ) ) {
+					string unauditedReason = BecauseHelpers.GetUnauditedReason( symbol );
+					return annotatedResult.WithSeenUnauditedReason( unauditedReason );
+				}

```

Comment:
I don't like doing this here. I would rather get all the `Because` reasons inside of `MutabilityInspectionResult.Annotated`. But I think that would require adjustments to the `BecauseHelper`, I'm thinking we can refactor that separately. Also I found that unlike statics you can have both the `Audited` and `Unaudited` mutability attribute on a single member.